<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Anna James-Bott" />

<meta name="date" content="2018-08-22" />

<title>R Markdown for RNA Seq analysis of AMO-CFZ cells</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.0.13/css/fa-svg-with-js.css" rel="stylesheet" />
<script src="site_libs/font-awesome-5.0.13/js/fontawesome-all.min.js"></script>
<script src="site_libs/font-awesome-5.0.13/js/fa-v4-shims.min.js"></script>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->






<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">RNA Seq AMO-CFZ</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li>
  <a href="maplot.html">MA Plots</a>
</li>
<li>
  <a href="pca_plots.html">PCA Plots</a>
</li>
<li>
  <a href="heat_maps.html">Heat Maps</a>
</li>
<li>
  <a href="xgr.html">XGR</a>
</li>
<li>
  <a href="resistance.html">Resistance</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">R Markdown for RNA Seq analysis of AMO-CFZ cells</h1>
<h4 class="author"><em>Anna James-Bott</em></h4>
<h4 class="date"><em>22 August 2018</em></h4>

</div>


<div id="hr-vs-24hr-differential-expression" class="section level1 tabset">
<h1>6hr vs 24hr differential expression</h1>
<p>Comparison of gene expression accross time points for the two more active compounds (NCP 22 and MAZ 1392)</p>
<div id="maz-13-6hr-vs-24hr" class="section level2">
<h2>MAZ 13 6hr vs 24hr</h2>
<p><img src="Figures/MAZ13_6hr_vs_24hr_foldchange.png" width="414" style="display: block; margin: auto;" /></p>
</div>
<div id="ncp-26-6hr-vs-24hr" class="section level2">
<h2>NCP 26 6hr vs 24hr</h2>
<p><img src="Figures/NCP26_6hr_vs_24hr_foldchange.png" width="414" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="section" class="section level1">
<h1></h1>
<p>Red points show genes which have increase/decrease over twofold (abs(x)&gt; 1 log2 FC) for both 6 hour and 24 hour time points.</p>
</div>
<div id="venn-diagram-of-genes-differentially-expressed" class="section level1 tabset">
<h1>Venn diagram of genes differentially expressed</h1>
<div id="maz13-6hr-vs-ncp26-6hr" class="section level2">
<h2>MAZ13-6hr vs NCP26-6hr</h2>
<p><img src="Figures/venn_6.png" width="662" style="display: block; margin: auto;" /> The intersection shows genes which are differentially expressed (padj &lt;0.0.5 and abs(foldchange) &gt;= 1) for both MAZ 13 and NCP 26 at the 6 hours time point. <br/> MAZ 13 (6hr) only - 70 genes<br/> NCP 26 (6hr) only - 14 genes<br/> Both (6hr)- 106 genes<br/> Total (6hr)- 190 genes<br/></p>
</div>
<div id="maz13-24hr-vs-ncp26--24hr" class="section level2">
<h2>MAZ13-24hr vs NCP26- 24hr</h2>
<p><img src="Figures/venn_24.png" width="663" style="display: block; margin: auto;" /> The intersection shows genes which are differentially expressed (abs(foldchange) &gt;= 1) for both MAZ 13 and NCP 26 at the 24 hours time point.<br/> MAZ 13 (24hr) only - 2 genes<br/> NCP 26 (24hr) only - 145 genes<br/> Both (24hr)- 63 genes<br/> Total (24hr)- 210 genes<br/></p>
</div>
</div>
<div id="section-1" class="section level1">
<h1></h1>
<p>At 6 hours more genes were differentially expressed by MAZ 13, but by the 24 hour time point, MAZ 13 has only 2 unique genes not differentially expressed by NCP 26, whilst NCP 26 at 24 hours has many more differentially expressed genes.</p>
</div>
<div id="venn-diagram-of-transcription-factors-differentially-expressed" class="section level1">
<h1>Venn diagram of transcription factors differentially expressed</h1>
<p><img src="Figures/venn_tf_4way.png" width="824" style="display: block; margin: auto;" /> 4 way cross of transcription factors with significant expression change (abs(log2 fc) &gt;=1 and padj &lt; 0.05 for 6hrs) compared to DMSO controls. <br/> TFcheckpoint used. Not all transcription factors included have experimental evidence that they are indeed TFs, but are possible TF candidates.</p>
</div>
<div id="aar-genes" class="section level1">
<h1>AAR genes</h1>
<p>Known genes involved in amino acid starvation response, expected to be upregulated <img src="Figures/AAR_genes.png" width="414" style="display: block; margin: auto;" /> Gene vs mean log2 fc taken across both time points. MAZ 13 shown in pink, NCP 26 shown in green.</p>
</div>
<div id="top-50-genes-with-largest-log-fold-change" class="section level1 tabset">
<h1>Top 50 genes with largest log fold change</h1>
<div id="ncp-26--6hr" class="section level2">
<h2>NCP 26- 6hr</h2>
<p><img src="index_files/figure-html/top_26_6-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="ncp-26--24hr" class="section level2">
<h2>NCP 26- 24hr</h2>
<p><img src="index_files/figure-html/top_26_24-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="maz-13--6hr" class="section level2">
<h2>MAZ 13- 6hr</h2>
<p><img src="index_files/figure-html/top_13_6-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="maz-13--24hr" class="section level2">
<h2>MAZ 13- 24hr</h2>
<p><img src="index_files/figure-html/top_13_24-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="maz-18--6hr" class="section level2">
<h2>MAZ 18- 6hr</h2>
<p><img src="index_files/figure-html/top_18_6-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="section-2" class="section level1">
<h1></h1>
<p>No NCP 22 genes or MAZ 18 (24 hours) were differentially expressed with the cut-off padj &lt; 0.05. For 24 hours, as one DMSO was an outlier, only two control values were being used as reference for differential expression. Therefore padj values were not used as a cut-off, instead abs(log2 fc) &gt; 1 has been used as a threshold.</p>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
